subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java

```diff
@@ -744,7 +744,7 @@ private DateTime alignToNextBCDIfRequired(final Plan curPlan, final PlanPhase cu
         }
 
         final BillingPeriod billingPeriod = curPlanPhase.getRecurring() != null ? curPlanPhase.getRecurring().getBillingPeriod() : BillingPeriod.NO_BILLING_PERIOD;
-        final LocalDate resultingLocalDate = BillCycleDayCalculator.alignProposedNextBillCycleDate(originalTransitionDate, bcd, billingPeriod, context);
+        final LocalDate resultingLocalDate = BillCycleDayCalculator.alignProposedNextBillCycleDate(originalTransitionDate, lastTransitionDate, bcd, billingPeriod, context);
```

Comment:
Unrelated to your change, but the name are so confusing. Would it make sense to rename
* `originalTransitionDate` -> `curTransitionDate` # Really the transition time associated with the `Plan#EffectiveDateForExistingSubscriptions`
* `lastTransitionDate` -> `prevTransitionDate`

And perhaps swap the 2 in the method signature?